docs/videos/healthcare-and-life-sciences/hcls-blueprint.yaml

```diff
@@ -0,0 +1,356 @@
+                  prefix: /usr/local
+              buildable: False
+      compilers: [gcc@9.3.0 target=x86_64]
+      environments:
+      - name: gromacs
+        type: file
+        content: |
+          spack:
+            definitions:
+            - compilers:
+              - gcc@9.3.0
+            - mpis:
+              - intel-mpi@2018.4.274
+            - mpi_packages:
+              - gromacs@2020.5
+            specs:
+            - matrix:
+              - - $mpis
+              - - $%compilers
+            - matrix:
+              - - $mpi_packages
+              - - $%compilers
+              - - $^mpis
+      # Un-comment and update mirror_url to install from spack cache
+      # spack_cache_url:
+      # - mirror_name: gcs_cache
+      #   mirror_url: gs://...
+
+  - id: spack-builder-startup  # TODO: Update example or make generic
+    source: modules/scripts/startup-script
+    settings:
+      runners:
+      - $(spack.install_spack_deps_runner)
+      - $(spack.install_spack_runner)
+      - type: shell
+        destination: setup_gromacs.sh
+        content: |
+          #!/bin/bash
+          source /apps/spack/share/spack/setup-env.sh
+          spack env activate gromacs
+          chmod -R a+rwX /apps/spack/var/spack/environments/gromacs
+          mkdir -p /apps/gromacs
+          chmod a+rwx /apps/gromacs
+          cd /apps/gromacs
+          wget --no-verbose https://ftp.gromacs.org/pub/benchmarks/water_GMX50_bare.tar.gz
+          tar xzf water_GMX50_bare.tar.gz
+
+      - type: data  # rethink
+        destination: /apps/gromacs/submit_gromacs.sh
+        content: |
+          #!/bin/bash
+          #SBATCH -N 2
+          #SBATCH --ntasks-per-node 30
+          #SBATCH -p compute
+
+          # Size can be 0000.65  0000.96  0001.5  0003  0006  0012  0024  0048  0096  0192  0384  0768  1536  3072
+          # Type can be 'pme' or 'rf'
+
+          source /apps/spack/share/spack/setup-env.sh
+          spack env activate gromacs
+
+          # Check that gmx_mpi exists
+          which gmx_mpi
+          cd $SLURM_SUBMIT_DIR
+          cp /apps/gromacs/water-cut1.0_GMX50_bare/1536/* .
+          scontrol show hostnames ${SLURM_JOB_NODELIST} > hostfile
+          gmx_mpi grompp -f pme.mdp -c conf.gro -p topol.top -o input.tpr
+          mpirun -n 60 -hostfile hostfile -ppn 30 gmx_mpi mdrun -notunepme -dlb yes -v -resethway -noconfout -nsteps 4000 -s input.tpr
+
+      - type: shell  # TODO: consider making bool option
+        destination: shutdown.sh
+        content: |
+          #!/bin/bash
+          if [ ! -f /etc/block_auto_shutdown ]; then
+                  touch /etc/block_auto_shutdown
+                  shutdown -h +1
+          fi
+
+  - id: spack-builder
+    source: modules/compute/vm-instance
+    use: [network-install, appsfs-install, spack-builder-startup]
+    settings:
+      name_prefix: spack-builder
+      machine_type: c2-standard-16  # TODO Consider bumping up, left low for quota
```

Comment:
In my experience, Spack will not go any faster with a larger machine. You can validate.
Also, I would remove the TODO comment.

Reply:
I have had similar experience when building from cache. Not sure about a true build from source. @douglasjacobsen do you have any input here?

Reply:
True build from source might end up being ram limited (unless you're using PD). It's possible to speed up a source build, using parallel builders, but also probably outside the scope of this blueprint.

Reply:
I have reduced to a c2-standard-4.

Reply:
Sorry, I was not clear. I have tried building from source with 2, 4, 8, 16, 32, 64 ... cores. Anything up to 16 gives benefits, beyond that point, it's not any faster. Spack only occupies 16 cores (at least the last time I have looked into this). Therefore, I would say 16 is the fastest machine you should pick. I would pick 16 and not try to do any bigger.

Reply:
Back to c2-standard-16.